docs/domain/domain_event/index.md

```diff
@@ -0,0 +1,76 @@
+# Domain event
+
+In Domain-Driven Design, a domain event is a pattern used to represent a significant change or occurrence within a domain. It is a way of capturing and communicating an event that has happened in the system and might be of interest to other parts of the system.
+
+## Packages
+
+| Package | Link | Application in section |
+| --- | ----------- | ----------- |
+| LeanCode.DomainModels | [![NuGet version (LeanCode.DomainModels)](https://img.shields.io/nuget/vpre/LeanCode.DomainModels.svg?style=flat-square&logo=nuget)](https://www.nuget.org/packages/LeanCode.DomainModels/8.0.2260-preview/) | `IAggregateRoot`, `IDomainEvent` |
+| LeanCode.DomainModels.Generators | [![NuGet version (LeanCode.DomainModels.Generators)](https://img.shields.io/nuget/vpre/LeanCode.DomainModels.Generators.svg?style=flat-square&logo=nuget)](https://www.nuget.org/packages/LeanCode.DomainModels.Generators/8.0.2260-preview/) | Ids |
+
+## Example
+
+Let's imagine that we want to perform some action after a employee has been assigned to a assignment. We will modify `AssignEmployeeToAssignment` method from the `Project` class:
```

Comment:
I would be somewhat wary of wording it like that. This somewhat states that the events should be created as a response to the need of doing something. This mental model suggests that you should raise an event whenever you need to do something, which might get twisted into events like `SendEmailToEmployee`.